test/functional/tests/seq_cutoff/test_seq_cutoff.py

```diff
@@ -0,0 +1,253 @@
+    NEGATIVE = auto()
+    POSITIVE = auto()
+    EQUAL = auto()
+
+
+@pytest.mark.parametrize("cache_mode, io_type, io_type_last", [
+    (CacheMode.WB, ReadWrite.write, ReadWrite.randwrite),
+    (CacheMode.WT, ReadWrite.write, ReadWrite.randwrite),
+    (CacheMode.WA, ReadWrite.read, ReadWrite.randread),
+    (CacheMode.WO, ReadWrite.write, ReadWrite.randwrite)])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_seq_cutoff_multi_core(cache_mode, io_type, io_type_last):
+    """
+    title: Sequential cut-off tests during sequential and random IO 'always' policy with 4 cores
+    description: |
+        Testing if amount of data written to cache after sequential writes for different
+        sequential cut-off thresholds on each core, while running sequential IO on 3 out of 4
+        cores and random IO against the last core, is correct.
+    pass_criteria:
+        - Amount of written blocks to cache is less or equal than amount set
+          with sequential cut-off threshold for three first cores.
+        - Amount of written blocks to cache is equal sequential to io size run against last core.
+    """
+    with TestRun.step("Test prepare (start cache and add cores)"):
+        cache, cores = prepare(cores_count=4)
+        writes_before = []
+        io_sizes = []
+        thresholds = []
+        fio_additional_size = Size(10, Unit.Blocks4096)
+        for i in range(len(cores)):
+            thresholds.append(Size(random.randint(1, int((cache.cache_device.size
+                                                          * 0.5).get_value(Unit.KibiByte)))))
+            io_sizes.append((thresholds[i] + fio_additional_size).align(0x1000))
+
+    with TestRun.step(f"Setting cache mode to {cache_mode}"):
+        cache.set_cache_mode(cache_mode)
+
+    for i, core in TestRun.iteration(enumerate(cores), "Set sequential cut-off parameters for "
+                                                       "all cores"):
+        with TestRun.step(f"Setting core sequential cut off policy to {SeqCutOffPolicy.always}"):
+            core.set_seq_cutoff_policy(SeqCutOffPolicy.always)
+
+        with TestRun.step(f"Setting core sequential cut off threshold to {thresholds[i]}"):
+            core.set_seq_cutoff_threshold(thresholds[i])
+
+    with TestRun.group("Preparing FIO configuration"):
+        fio = (Fio().create_command()
+               .io_engine(IoEngine.libaio)
+               .direct())
+
+        # Run sequential IO against first three cores
+        for i, core in TestRun.iteration(enumerate(cores[:-1])):
+            fio_job = fio.add_job(job_name=f"core_{core.core_id}")
+            fio_job.size(io_sizes[i])
+            fio_job.read_write(io_type)
+            fio_job.target(core.system_path)
+            writes_before.append(core.get_core_statistics(
+                stat_filter=[StatsFilter.blk])["writes to cache"])
```

Comment:
This should be within step. BTW you don't need to use `TestRun.iteration()` everywhere. You can embed normal loop within step, especially when printing every iteration of the loop doesn't bring much information.